sdcm/docker.py

```diff
@@ -141,27 +146,38 @@ def __init__(self, **kwargs):
 
     def _create_node_image(self):
         self._update_image()
+        private_key = self.credentials[0].key_file
+        context_pub_key_path = os.path.join(self._context_path, 'scylla-test.pub')
+        LOCALRUNNER.run(f"ssh-keygen -y -f {private_key} > {context_pub_key_path}; sed -ri "
+                        f"'s/(ssh-rsa [^\\n]+)/\\1 test@localhost/g' {context_pub_key_path}")
         _cmd(f'build --build-arg SOURCE_IMAGE={self._image}:{self._version_tag} -t {self._node_img_tag} {self._context_path}',
              timeout=300)
+        LOCALRUNNER.run(f'rm -f {context_pub_key_path}', ignore_status=True)
 
     @staticmethod
-    def _clean_old_images():
+    def _clean_not_user_containers():
+        _cmd('container prune -f')
+
+    def _clean_old_images(self):
         images = " ".join(_cmd('images -f "dangling=true" -q').stdout.split('\n'))
         if images:
+            self._clean_not_user_containers()  # Remove old containers that could block removing dangling images
             _cmd(f'rmi {images}', timeout=90)
 
     def _update_image(self):
         LOGGER.debug('update scylla image')
         _cmd(f'pull {self._image}:{self._version_tag}', timeout=300)
-        self._clean_old_images()
+        try:
+            self._clean_old_images()
```

Comment:
`docker system prune --volumes -f`
cleans:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - all dangling build cache